sgt-back-end/index.js

```diff
@@ -0,0 +1,156 @@
+const express = require('express');
+const app = express();
+
+const pg = require('pg');
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+app.get('/api/grades', (req, res) => {
+  const sql = `
+  select *
+    from "grades"
+  `;
+  db.query(sql)
+    .then(result => {
+      const grades = result.rows;
+      res.json(grades);
+    })
+    .catch(err => {
+      // eslint-disable-next-line no-console
+      console.log(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+app.use(express.json());
+
+app.post('/api/grades', (req, res) => {
+  const score = Number(req.body.score);
+  const params = [req.body.name, req.body.course, req.body.score];
+  if (req.body.name === undefined || req.body.course === undefined || req.body.score === undefined) {
+    res.status(400).json({
+      error: 'Invalid grade. Name, course, and score are required fields.'
+    });
+  }
+  if (!Number.isInteger(score) || score < 0 || score > 100) {
+    res.status(400).json({
+      error: 'Invalid grade. Score must be an integer between 0 and 100.'
+    });
+  }
```

Comment:
Calling `res.json` doesn't prevent any future `res` methods from being called in the same middleware. Please make sure only one response can be sent in any given route. Also, if a failure response has already been sent, it's not a good idea to continue with the database update.